File: Leetcode/PY/Leetcode74-搜索矩阵.py
from typing import List
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n,m = len(matrix),len(matrix[0])
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == target:
                    return True
                elif matrix[i][j] > target:
                    return False
        return False

File: Leetcode/Rust/Leetcode74-搜索矩阵.rs
impl Solution {
    pub fn search_matrix(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        let (n,m) = (matrix.len(),matrix[0].len());
        for i in 0..n{
            for j in 0..m{
                if matrix[i][j] == target{
                    return true;
                }else if matrix[i][j] > target{
                    return false; 
                }
            }
        }
        false
    }
}